### PR TITLE
[2/N] executor pass the complete config to worker/modelrunner

### PR DESCRIPTION
**Pull request by youkaichao**
continue of https://github.com/vllm-project/vllm/pull/9933

### CODE REVIEW

File: vllm/worker/multi_step_worker.py [hunk 1]
```diff
@@ -25,20 +25,7 @@ def __init__(self, *args, **kwargs):
...
-            base_model_runner.device_config,
-            base_model_runner.cache_config,
-            load_config=base_model_runner.load_config,
-            lora_config=self.lora_config,
-            kv_cache_dtype=self.cache_config.cache_dtype,
-            is_driver_worker=base_model_runner.is_driver_worker,
-            prompt_adapter_config=base_model_runner.prompt_adapter_config,
-            observability_config=base_model_runner.observability_config,
-        )
+        self.model_runner = MultiStepModelRunner(base_model_runner, )
```
Comment (njhill) [inline vllm/worker/multi_step_worker.py:28]: nit:
```suggestion
        self.model_runner = MultiStepModelRunner(base_model_runner)
```

File: vllm/worker/neuron_model_runner.py [hunk 2]
```diff
@@ -57,20 +56,13 @@ class NeuronModelRunner(ModelRunnerBase[ModelInputForNeuron]):
 
     def __init__(
         self,
-        model_config: ModelConfig,
-        parallel_config: ParallelConfig,
-        scheduler_config: SchedulerConfig,
-        device_config: DeviceConfig,
+        vllm_config: ModelConfig,
```
Comment (robertgshaw2-neuralmagic) [inline vllm/worker/neuron_model_runner.py:59]: Is this meant to be `EngineConfig`?

Comment (njhill) [inline vllm/worker/neuron_model_runner.py:59]:
```suggestion
        vllm_config: EngineConfig,
```

File: vllm/worker/neuron_worker.py [hunk 3]
```diff
@@ -44,7 +36,7 @@ def __init__(
             init_cached_hf_modules()
 
         self.model_runner: NeuronModelRunner = NeuronModelRunner(
-            model_config, parallel_config, scheduler_config, device_config)
+            vllm_config=vllm_config, )
```
Comment (njhill) [inline vllm/worker/neuron_worker.py:39]:
```suggestion
            vllm_config=vllm_config)
```

**Comment by robertgshaw2-neuralmagic** [comment]
I personally find it confusing that we call the members/arguments `vllm_config` when it is of type `EngineConfig`. I would personally choose the name `engine_config`, but I dont have religion around it.

Other than the `ModelConfig` --> `EngineConfig` comment, approving to unblock

**Review by robertgshaw2-redhat — APPROVED** [review]

**Review by njhill — COMMENTED** [review]
Nice cleanup!

**Comment by njhill** [comment]
> I personally find it confusing that we call the members/arguments `vllm_config` when it is of type `EngineConfig`. I would personally choose the name `engine_config`, but I dont have religion around it.

Agree .. alternatively `EngineConfig` could be renamed to `VLLMConfig` :)

File: vllm/worker/neuron_model_runner.py (hunk 2 above)
Comment (youkaichao) [reply vllm/worker/neuron_model_runner.py:59]: nice catch!

Comment (youkaichao) [reply vllm/worker/neuron_model_runner.py]: fixed in [02242c7](https://github.com/vllm-project/vllm/pull/9938/commits/02242c7d5e0fa794b4f05ca97b4dc1e93356f250)

**Comment by youkaichao** [comment]
@robertgshaw2-neuralmagic @njhill renamed to `VllmConfig`  👍

**Pull request merged**